### DISCUSSION TITLE
Parking type: layby

### DISCUSSION THREAD

**Discussion by drazde** [streetcomplete/StreetComplete#5704] [General]
Is it possible add _layby_ like type of parking area?

~~~
amenity=parking
parking=layby
~~~

https://wiki.openstreetmap.org/wiki/Tag%3Aparking%3Dlayby

Thanks

**Comment by matkoniecz** [comment]
refers likely to https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/parking_type/ParkingType.kt

how it differs from `parking=street_side` ? After quick reading wiki page it is not clear at all to me.

**Reply by drazde** [reply]
For me _layby_  is not really a marked parking.
Is a space near the street that you can leave the car occasionally, for few hour not a lot.
Mainly on secondary or mountain roads

**Reply by matkoniecz** [reply]
So layby would be for entire area, and have streetside parking (or parking lot) inside?

**Reply by drazde** [reply]
For a small area streetside

**Reply by jdhoek** [reply]
Wrong. `street_side` and `layby` are both for small areas where cars can park. **They can look exactly the same, but have radically different functions.** A `layby` is a small area alongside a larger road outside of city limits where motorists can safely stop and rest in case of fatigue or other inconveniences. See the photos on the wiki for what these typically are. These are important for road safety, and a good motorists map will render them prominently. These are not as big as `highway=rest_area`. If you have a rest area with its own access road, bins, and pick-nick tables, then you are likely looking at `highway=rest_area`, and you won't use `parking=layby` for the individual parking bays (you could use `street_side` there).

Sometimes lay-bys also fulfil the role of street-side parking on lower level roads. Mappers can judge for themselves if these should still be classified as lay-bys or just as street-side parking. Something like this though:

![Parking_layby_France](https://github.com/streetcomplete/StreetComplete/assets/683699/80db5b37-55ef-48b1-b99a-2d06b00f2c52)

This is never street-side parking; it's a lay-by. Parking here is only useful if you need to rest and then move on. You wouldn't park here and go to visit some shop kilometres away.

**Reply by matkoniecz** [reply]
It would be helpful if this info (if that is a general shared view by mappers) would be on the OSM Wiki page for this tag.

**Reply by mnalis** [reply]
- Does the [parking=layby](https://wiki.openstreetmap.org/wiki/Tag:parking%3Dlayby) fulfills all the following 3 conditions, or does it fail some of them? 
  - Area suitable or designated for parking 
  - directly adjacent to the carriageway of a road 
  - can be reached directly from the roadway (without having to use an access way)

  If it fulfills all of them, then it seems to be a more precise sub-type of more general [parking=street_side](https://wiki.openstreetmap.org/wiki/Tag:parking%3Dstreet_side), right? (note that OSM tag `parking=street_side` might have a somewhat **different meaning** than English words _"Street side parking"_ - one has to read its wiki to understand what is covered by it).

- `parking=layby` currently has some 10k uses worldwide, which is about **0.31%** of `parking=*` values.  That is,  there is less than 1 in 300 chance that it is the answer user is seeking. Perhaps other answer of leaving a note (already existing) would suffice for such relatively rare occurrence? As with other quests, there is no way we can offer [all 819 values of parking](https://taginfo.openstreetmap.org/keys/parking#values) - just the most popular ones.

- Below is picture how to the quest currently looks. If `parking=layby` is still deemed important as a separate answer, how it would need to change? Especially, what picture and text should it have, but also how other pictures and texts would need to change so that e.g. pedestrian which has never owned a car is not confused which answer to choose (i.e. _"👨‍💻 Users are no experts"_ and _"🐿️ Easy answer"_ [quest guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md)) ? Especially given the explanation that ***"They can look exactly the same, but have radically different functions"*** 
![small_Screenshot_20240626_165532_SCEE](https://github.com/streetcomplete/StreetComplete/assets/156656/3049e1de-5346-4ce5-9297-b61b90a19018)


- if the basic [verifiable](https://wiki.openstreetmap.org/wiki/Verifiability) difference between them is that `parking=layby` is on rural highway far from amenities, and `parking=street_side` is for ones inside a cities (which is my best understanding at the moment; please clarify if I got it wrong); is StreetComplete best place to do those edits, or would it be easier to just [(re-)tag](https://wiki.openstreetmap.org/wiki/Mechanical_Edits) as such in JOSM/iD? (see _"💵 Valuable Surveyors"_ quest guideline)

**Reply by jdhoek** [reply]
No, lay-bys differ from street-side parking in that they are not generally speaking very suitable for all-purpose parking. They are really only suitable — and in that case highly desirable — for parking by travellers en-route needing an urgent break. **They have different primary uses.**

[We've kept this `parking`-value as such in the `street_side`-proposal.](https://wiki.openstreetmap.org/w/index.php?oldid=2066642) If you really feel getting rid of it is better than handling it properly, you should draft a proposal on the wiki. What should not happen is that StreetComplete users start retagging these as street-side parking.

**Reply by mnalis** [reply]
> They have different primary uses.

I read you. But how can one [verifiably](https://wiki.openstreetmap.org/wiki/Verifiability) **determine** ***which*** of the primary purposes it has? Is there a special sign? Some visual difference? Or just its location (e.g. inside or outside of city limits) / proximity to the amenities? Or is it determined in some other way? How **exactly**?

> [We've kept this parking-value as such in the street_side-proposal.](https://wiki.openstreetmap.org/w/index.php?oldid=2066642)

Thanks for the link. It seems then that somebody put incorrect [status](https://wiki.openstreetmap.org/wiki/Tag_status) for that value, both at [Key:parking](https://wiki.openstreetmap.org/wiki/Key:parking) and [Tag:parking=layby](https://wiki.openstreetmap.org/wiki/Tag:parking=layby). If that value was indeed voted on and approved, it's status should be `Approved`, not `In use`.  Would you care to fix it?

> If you really feel getting rid of it is better than handling it properly, you should draft a proposal on the wiki.

I am not trying to get rid of it, I'm **trying to understand what it is and how would I recognize one when I see it**. 
(note that I'm not familiar with the terms, am not a native English speaker, and don't own a car. I'm prolific SC user, though).

The current wiki and explanations provided so far do not seem to provide me with enough information to make a distinction. Apparently, they can look exactly the same, so I cannot determine which one it is just by the looks of it?
You seem to be saying that my guess that it can be determined by it being inside/outside of the city limits is also not correct? Or am I misunderstanding your "No"? The parking sign from the pictures also seems to look identical, and I do not see some special marks on the floor either.
 
**How** then do I reliably determine whether it is `street_side` or `layby`? I do not own a crystal ball, and thus cannot read other people minds what was **"intended primary use"** of some parking  (unless it is signed somewhere, of course!)
In other words, I would need some **physical** evidence I could check and which would give me definite answer whether a specific instance of parking  is `street_side` ***or*** `layby`.

> What should not happen is that StreetComplete users start retagging these as street-side parking.

Are we still talking about [SC "What kind of parking is this?" Quest](https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/parking_type/ParkingType.kt) here? It should only ask for `amenity=parking` which do not have `parking=*` value (or have a `parking=yes`). So, it should not touch any parkings already tagged with `parking=layby`. Or do you have different experience?

**Reply by jdhoek** [reply]
I've edited the lay-by wiki page to add some points people can use to determine whether something is a lay-by or not. The problem with a tag like this is that while it has a clear meaning to anyone with a driver's licence, it is also old and not really looked after. I have documented it as collateral of `parking=street_side` when we introduced that tag, but it wasn't part of what was voted on; meanwhile the tag's status was changed to de-facto, which seems correct.

> Apparently, they can look exactly the same, so I cannot determine which one it is just by the looks of it?

Nope. It's all about the context of the object. Fortunately this can be determined fairly well from the surroundings.

**Reply by matkoniecz** [reply]
I am still confused and commented on https://wiki.openstreetmap.org/wiki/Talk:Tag:parking%3Dlayby#Instead
